src/Microsoft.ML.Data/Scorers/MultiClassClassifierScorer.cs

```diff
@@ -379,7 +379,7 @@ private static ISchemaBoundMapper WrapIfNeeded(IHostEnvironment env, ISchemaBoun
 
             if (trainSchema?.Label == null)
                 return mapper; // We don't even have a label identified in a training schema.
-            var keyType = trainSchema.Label.Value.Metadata.Schema.GetColumnOrNull(MetadataUtils.Kinds.KeyValues)?.Type;
+            var keyType = trainSchema.Label.Value.Metadata.Schema.GetColumnOrNull(MetadataUtils.Kinds.KeyValues)?.Type as VectorType;
```

Comment:
>as VectorType; [](start = 119, length = 14)

Can you point on a reason for this new constraint?

Reply:
Same reasoning as for BinaryClassifierScorer.cs - 

`CanWrap` checks whether this column is a `VectorType`, so this is "safe". Doing the cast was necessary for line 71's usage of `keyType.ItemType`.

---
In reply to: [247769535](https://github.com/dotnet/machinelearning/pull/2131#discussion_r247769535) [](ancestors = 247769535)